internal/pkg/test/cmd.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package test
+
+import (
+	"testing"
+
+	"gotest.tools/icmd"
+)
+
+type Cmd struct {
+	path string
+}
+
+type Result struct {
+	*icmd.Result
+}
+
+func NewCmd(path string) *Cmd {
+	c := &Cmd{
+		path: path,
+	}
+
+	return c
+}
+
+func (c *Cmd) Run(t *testing.T, args ...string) *Result {
+	t.Logf("Running cmd %s with args %s", c.path, args)
```

Comment:
We can use some quotes here, just to make it clear what's running and how it's running.

What I'm thinking is that if you are going to run something like `singularity foo --bar=baz xyz` you want to be able to differentiate flag `--bar=baz` with extra argument `xyz` from flag `--bar=baz xyz`.

The quotes around c.path are trivial, just add them to the format string.

The quotes around the arguments...

A very basic way of handling this is something like:

```
type stringQuoter []string

func (sq stringQuoter) Format(f State, c rune) {
    sep := []byte{}
    commaBlank := []byte{',', ' '}
    for _, s := range sq {
        f.Write(sep)
        f.Write([]byte(`"`))
        f.Write([]byte(s))
        f.Write([]byte(`"`))
        sep = commaBlank
    }
}
```

and then you wrap `args` in that type when passing it to `t.Logf`.

(and I'm sure that code can be much much better, I'm just providing it as a general idea of how this could be done)

Reply:
To quote arguments, I switched from `%s` to `%q`.

Does this seem OK to you?

`cmd.go:27: Running cmd "/usr/local/singularity/bin/singularity" with args ["help" "oci" "pause"]`

Reply:
That's good enough for this